src/components/BottomBar/UserControlPanel/UserControlPanel.jsx

```diff
@@ -0,0 +1,148 @@
+import Button from '../../common/Input/Button/Button';
+import Select from '../../common/Input/Select/Select';
+import Popover from '../../common/Popover/Popover';
+import Row from '../../common/Row/Row';
+import Input from '../../common/Input/Input/Input';
+
+import Picker from '../Picker';
+
+import styles from './UserControlPanel.scss';
+
+function UserControlPanel() {
+  const [fileList, setFileList] = React.useState(undefined);
+  const [selectedPanel, setSelectedPanel] = React.useState(undefined)
+  const [panelURL, setPanelURL] = React.useState(undefined)
+  const popoverVisible = useSelector(
+    (state) => state.local.popovers.userControlPanel.visible
+  );
+  const luaApi = useSelector((state) => state.luaApi);
+  const isDataInitialized = useSelector((state) => state.userPanels.isInitialized);
+
+  const panelList = useSelector((state) => state.userPanels.panels || []);
+
+  const dispatch = useDispatch();
+    if (luaApi && !isDataInitialized) {
+        dispatch(loadUserPanelData(luaApi));
+    }
+
+  
+  function togglePopover() {
+    //Todo MICAH this was to avoid creating a topic
+    //Didnt feel right making a topic just for this
+    //Thinking to adda a generic folder watching topic 
+    if (!popoverVisible) {
+      dispatch(loadUserPanelData(luaApi));
+    }
+    dispatch(setPopoverVisibility({
+      popover: 'userControlPanel',
+      visible: !popoverVisible
+    }));
+  }
+
+  function updatePanelSelection(selection) {
+    setSelectedPanel(selection.value);
+  }
+
+  function addPanel() {
+    dispatch(addUserPanel(selectedPanel))
+  }
+
+  function addWebPanel() {
+    dispatch(addUserPanel(panelURL))
+  }
+
+  function updatePanelURL(evt) {
+    setPanelURL(evt.target.value)
+  }
+
+  function popover() {
+
+    var fileList = ["suncontroler", "buttonpage"];
+    var placeholderText = "Loading pages"
+
+    const options = Object.values(panelList)
+      .map((panel) => ({ value: panel.path, label: panel.name }));
+
+      return (
+      <Popover
+        className={Picker.Popover}
+        title="User Control Panels"
+        closeCallback={() => togglePopover()}
+        detachable
+        attached
+      >
+        <div className={Popover.styles.content}>
+          <Row>
+          <Select
+              menuPlacement="top"
+              placeholder={placeholderText}
+              options={options}
+              label="Select Panel"
+              onChange={updatePanelSelection}
+              value={selectedPanel}
+            />
+            
+            <div className={Popover.styles.row}>
+              <Button
+                onClick={addPanel}
+                title="Add panel"
+                style={{ width: 90 }}
+                disabled={!selectedPanel}
+              >
+                <MdWeb alt="add panel" />
+                <span style={{ marginLeft: 5 }}>Add Panel</span>
+              </Button>
+            </div>
+          </Row>
+          <hr className={Popover.styles.delimiter} />
```

Comment:
Just a note that this has bene replaced with a component in another PR, so when that one is merged this will break. FYI :) 

https://github.com/OpenSpace/OpenSpace-WebGuiFrontend/pull/174/files